### ISSUE TITLE
Docker image doesn't work

### ISSUE THREAD

**Issue by forki** [fable-compiler/fable-suave-scaffold#42]
We have a very simple Dockerfile that just tries to dockerize an already "published" .NETCore app:

    FROM microsoft/dotnet:1.1.0-sdk-msbuild-rc4
    COPY . .
    ENTRYPOINT ["dotnet", "Server.dll"]

you can create the image with:

    build.cmd publish

If you then run the container with:

    docker run -it --rm forki/fable-suave

It fails with the error:

    Error: assembly specified in the dependencies manifest was not found -- package: 'fsharp.core', version: '4.1.0', path: 'lib/netstandard1.6/FSharp.Core.dll'

This is technically correct, but I you do the following it works:

    cd deploy
    dotnet Server.dll

So something inside the Dockerfile is causing the runtime to look at different place.

**Comment by Petermarcu** [comment]
Are you publishing the application? Can you share your steps for building the application (`server.dll`)?

**Comment by forki** [comment]
@Petermarcu  as written above `build.cmd publish`

**Comment by forki** [comment]
build.sh publish on Linux/MacOs

**Comment by Petermarcu** [comment]
Ok, I missed that. Looks like you are running `dotnet publish` as expected. You also have the `PackageReference` to the FSharp SDK in your Server project.

You say that it runs from the `deploy`. That is what I'd expect since that is where you seem to have specified to publish the application to? The contents of your `deploy` dir are what you should be copying into the image and executing.

**Comment by mastoj** [comment]
Maybe this might help: https://github.com/mastoj/suavecore

A small sample app running suave on core on docker I made.

**Comment by forki** [comment]
@mastoj can you see anything that I might be doing wrong?

**Comment by forki** [comment]
@petermarcu I copy the full deploy dir into the container. (first I copy the Dockerfile to /deploy and then I run docker build  over there)

**Comment by forki** [comment]
@Petermarcu I pushed the docker image to dockerhub at forki/fable-suave if you want to inspect the end result

**Comment by forki** [comment]
Summoning @condrong (@shanselman told me you might know)

**Comment by mastoj** [comment]
@forki, can't spot anything obvious. If it helps, here are the commands I run: https://github.com/mastoj/suavecore/blob/master/commands

**Comment by forki** [comment]
@mastoj why do you have dotnet run before the docker build? Needed?

**Comment by forki** [comment]
Also I'm using rc4 of the dotnet cli with MSBuild. So no project.json

**Comment by mastoj** [comment]
@forki I see that the "script" might be a little bit confusing, that's because it is not a script just a file with commands that I used while playing around.

**Comment by forki** [comment]
So it's not needed and it just work's because it's project.json and we now
have a new bug in current rc4?

Am 21.02.2017 09:14 schrieb "Tomas Jansson" <notifications@github.com>:

**Comment by mastoj** [comment]
That I don't know. I can't see what the problem is, mainly because I can't build the project on OSX. It shouldn't make a difference if you have project.json, at least that's what I thought. 

What I needed to do, when I played with it was this (if I recall correctly):

```
dotnet build
dotnet publish -c Release
```

The I used the published bits in the container:

```
FROM microsoft/dotnet:core
COPY ./bin/Release/netcoreapp1.0/publish /app
WORKDIR /app
EXPOSE 8083
ENTRYPOINT ["dotnet", "suavecore.dll"]
```

Of course I struggled to get the correct .NET Core  version, tooling etc. to get it to work, but after that it worked like a charm.
